### PR TITLE
KB-15093 | DEV |BE | Implementation of the APIs for creating the Training Plans

### PR DESCRIPTION
**Pull request by tarentomaheshvakkund**
Introduces the CB Plan V3 API under a new `com.igot.cb.cbplan` package, alongside the existing V1/V2 controllers. No existing endpoints changed or removed.

### New API — `/cbplan/v3`
Method	Path	Purpose
POST	/create	Create a CB Plan (DRAFT)
POST	/update	Update DRAFT in place; LIVE plans staged into draftData
POST	/publish	Publish DRAFT→LIVE, or re-publish a staged draft
DELETE	/archive	Retire a plan and deactivate its org lookups
GET	/read/{cbPlanId}	Read a plan with enriched content metadata
POST	/search	ES-backed search with createdBy/content enrichment
POST	/user/dictionary	Per-user content dictionary grouped by APAR/non-APAR

### Structure
The V2 god-class is split into focused, independently testable collaborators — validation, data transform, org lookup, content lookup, ElasticSearch and enrichment — orchestrated by `CbPlanServiceV3Impl`.

### Caching
- `CbPlanCacheMgrV3`: two-tier Caffeine cache (year-scoped lookups + per-planId), configurable TTL/size, with batched Cassandra reads instead of per-ID queries.
- `/user/dictionary` results cached in Redis per user+planYear with explicit TTL.
- User basic profile read from the shared `user:basicProfile:{userId}` key written by cb-ext-userprofile-service, falling back to Cassandra on a miss.
- Content metadata reuses the `extended_read_content_{id}` key, falling back to the extended/external content read APIs.

### Publish durability
Publish uses the pre-commit-validator + rollback overload of `updateRecord`, so the ES document is written before the Cassandra commit and rolled back if the commit fails. Unrecoverable divergence is logged as `ES_CASSANDRA_DIVERGENCE`.

### Config
New `cbplan.enriched.content.fields`, `cb.plan.v3.*` and `cassandra.query.limit.*` properties; `planYear` added to the CB Plan ES mapping; V3 table names, error messages, request/response keys and cache-key prefixes extracted into `Constants`.

### Tests
254 unit tests (JUnit 5 + Mockito), all passing — all 7 endpoints, service happy/error paths, `CbPlanCacheMgrV3` cache and batching behaviour, collaborator services, DTOs, and `CbPlanYearUtil` financial-year edge cases.

### CODE REVIEW

File: src/main/java/com/igot/cb/cbplan/service/impl/CbPlanServiceV3Impl.java [hunk 1]
```diff
@@ -0,0 +1,1519 @@
...
+        } else {
+            response.getParams().setStatus(Constants.FAILED);
+            response.getParams().setErr("CbPlan is in invalid state for publish. Status: " + existingStatus);
+            response.setResponseCode(HttpStatus.BAD_REQUEST);
+            return Collections.emptyMap();
+        }
+    }
+
+    private Map<String, Object> prepareCbPlanForRePublish(Map<String, Object> existingCbPlan,
+                                                          Map<String, Object> incomingRequest) throws JsonProcessingException {
```
Comment (ramyaranganathan-1775) [inline src/main/java/com/igot/cb/cbplan/service/impl/CbPlanServiceV3Impl.java:414]: prepareCbPlanForRePublish existing attributes of this implementation covered ? 

isApar ? orgScope ? please verify against  existing impl

Comment (tarentomaheshvakkund) [reply src/main/java/com/igot/cb/cbplan/service/impl/CbPlanServiceV3Impl.java:414]: Existing APIs are just refractored - there is no change in business logic , this need to be validated as part of dev testing.
